torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -740,13 +741,13 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
     }
 
     auto shouldUpdateMinEndTimePredicate = [&, this]() -> bool {
-      // Notice, whoever modifying `rpcAgentRunning_`
-      // must acquire lock on `futureMutex_`.
+      // Notice, whoever modifies `timeoutThreadEnabled_`
+      // must acquire a lock on `futureMutex_`.
       // Otherwise, this predicate could deadlock.
       // If during evaluating the predicate, `::shutdown()` is called, then
       // the predicate missed the notification before it started waiting
       // on the cond var.
-      if (!rpcAgentRunning_.load()) {
+      if (!timeoutThreadEnabled_.load()) {
```

Comment:
maybe you also update rpcAgentRunning_ in line 730 as well to this var?

(technically it would work as-is, but if the underlying assumption ever changes, we end up with a busy spin here)